liquibase-core/src/test/java/liquibase/parser/core/yaml/YamlSnapshotParserTest.java

```diff
@@ -0,0 +1,207 @@
+        String invalidYaml = "snapshot:\n  - invalid: [unclosed bracket";
+        ResourceAccessor resourceAccessor = createMockResourceAccessor("test.yaml", invalidYaml);
+
+        LiquibaseParseException exception = assertThrows(LiquibaseParseException.class, () -> {
+            parser.parse("test.yaml", resourceAccessor);
+        });
+
+        assertTrue(exception.getMessage().contains("Syntax error"));
+    }
+
+    @Test
+    void parse_withEmptyFile_throwsLiquibaseParseException() {
+        String emptyContent = "";
+        ResourceAccessor resourceAccessor = createMockResourceAccessor("test.yaml", emptyContent);
+
+        assertThrows(LiquibaseParseException.class, () -> {
+            parser.parse("test.yaml", resourceAccessor);
+        });
+    }
+
+    @Test
+    void parse_withValidSnapshot_returnsDatabaseSnapshot() {
+        String validSnapshot = "snapshot:\n" +
+                "  database:\n" +
+                "    shortName: h2\n" +
+                "  objects: {}\n";
+        ResourceAccessor resourceAccessor = createMockResourceAccessor("test.yaml", validSnapshot);
+
+        try {
+            DatabaseSnapshot result = parser.parse("test.yaml", resourceAccessor);
+            assertNotNull(result);
+        } catch (LiquibaseParseException e) {
+            assertFalse(e.getMessage().contains("Syntax error"));
+        }
+    }
+
+    @Test
+    void parse_withMetadata_parsesMetadataSuccessfully() {
+        String snapshotWithMetadata = "snapshot:\n" +
+                "  database:\n" +
+                "    shortName: h2\n" +
+                "  metadata:\n" +
+                "    key1: value1\n" +
+                "    key2: value2\n" +
+                "  objects: {}\n";
+        ResourceAccessor resourceAccessor = createMockResourceAccessor("test.yaml", snapshotWithMetadata);
+
+        try {
+            DatabaseSnapshot result = parser.parse("test.yaml", resourceAccessor);
+            assertNotNull(result);
+            assertNotNull(result.getMetadata());
+        } catch (LiquibaseParseException e) {
+            assertFalse(e.getMessage().contains("Syntax error"));
+        }
+    }
+
+    @Test
+    void parse_withLargeContent_doesNotFailDueToCodePointLimit() {
+        StringBuilder largeContent = new StringBuilder();
+        largeContent.append("snapshot:\n");
+        largeContent.append("  database:\n");
+        largeContent.append("    shortName: h2\n");
+        largeContent.append("  objects:\n");
+
+        String paddedValue = String.join("", Collections.nCopies(100, "x"));
+        for (int i = 0; i < 35000; i++) {
+            largeContent.append("    key").append(i).append(": ").append(paddedValue).append("\n");
+        }
+
+        ResourceAccessor resourceAccessor = createMockResourceAccessor("large-snapshot.yaml", largeContent.toString());
+
+        try {
+            parser.parse("large-snapshot.yaml", resourceAccessor);
```

Comment:
1. Can we use assertThorws for hanling this case?
2. I think that we need to assert what is an actual error message to make sure that it is not about limits, but with different error message